Xext/vidmode.c

```diff
@@ -1754,7 +1741,8 @@ ProcVidModeGetPermissions(ClientPtr client)
     if (client->swapped)
         swaps(&stuff->screen);
 
-    if (stuff->screen >= screenInfo.numScreens)
+    ScreenPtr pScreen = dixGetScreenPtr(stuff->screen);
+    if (!pScreen)
```

Comment:
We don’t really need the `pScreen` variable below this point so maybe just do `if (!dixGetScreenPtr(stuff->screen))` or keep the old version?

Reply:
Or maybe introduce a function or a macro `dixScreenExists(idx)` for better readability. There are several instances below where it could be used.